internal/api/v1/testcases.go

```diff
@@ -204,8 +218,18 @@ func UpdateTestCase(testCaseService services.TestCaseService, logger logging.Log
 //	@Failure		400			{object}	problemdetail.ProblemDetail
 //	@Failure		500			{object}	problemdetail.ProblemDetail
 //	@Router			/v1/test-cases/{testCaseID} [delete]
-func DeleteTestCase(services.TestCaseService) fiber.Handler {
+func DeleteTestCase(testCaseService services.TestCaseService) fiber.Handler {
 	return func(c *fiber.Ctx) error {
-		return problemdetail.NotImplemented(c, "failed to delete TestCase")
+		testCaseIDParam := c.Params("testCaseID")
+
+		err := testCaseService.DeleteByID(c.Context(), testCaseIDParam)
+		if err != nil {
+			return problemdetail.BadRequest(c, "failed to delete page")
+		}
```

Comment:
Should be ServerError and add logging. Although, depending on whether the delete can actually be performed or not you may want to consider having a check for what kind of error is actually returned using `errors.Is`

Reply:
@Wakisa maybe try re-pushing changes - this hasn't changed.